magento/magento2
# magento/magento2#35310: Rest API response "page_size" missing in M2.4.4

- fixed issue with missing "page_size" in response





### Description (*)
The problem was due to the fact that when the method processSearchCriteria in the class Magento\Framework\Webapi\Validator\IOLimit\DefaultPageSizeSetter try to get the default value for page size, it first checked if the configuration Enable Input Limits was enabled, but it not correct for my opinion, because it's doesn't relate to default page size values, so I removed this condition.

### Related Pull Requests

### Fixed Issues (if relevant)
1. Fixes magento/magento2#35310

### Manual testing scenarios (*)
There are four scenarios to check:
1. Set the Default Page Size value in  admin configuration:
    - Go to the `Stores->Configuration->Services->Magento Web API->Web Api Input Limits and set the Default Page Size value `
    (how example 500)
    - Create admin integration token using POST: `/rest/default/V1/integration/admin/token`
    - Use the API endpoint `/rest/all/V1/orders/?searchCriteria[filterGroups][0][filters][0][field]=status&searchCriteria[filterGroups][0][filters][0][value]=processing&searchCriteria[filterGroups][0][filters][0][conditionType]=eq&searchCriteria[currentPage]=1`
    - This should return response like:
    

```
    {
    "items": [],
    "search_criteria": {
        "filter_groups": [
            {
                "filters": [
                    {
                        "field": "status",
                        "value": "processing",
                        "condition_type": "eq"
                    }
                ]
            }
        ],
        "page_size": 500,
        "current_page": 1
    },
    "total_count": 0
}
```


So in "page_size" we see value from admin configurations

2. Use the same API endpoint but with "page_size" param (to check that API retrieve custom params correctly):
So request with the API endpoint rest/all/V1/orders/?searchCriteria[filterGroups][0][filters][0][field]=status&searchCriteria[filterGroups][0][filters][0][value]=processing&searchCriteria[filterGroups][0][filters][0][conditionType]=eq&searchCriteria[currentPage]=1&searchCriteria[pageSize]=20 must return response like this:

```
{
    "items": [],
    "search_criteria": {
        "filter_groups": [
            {
                "filters": [
                    {
                        "field": "status",
                        "value": "processing",
                        "condition_type": "eq"
                    }
                ]
            }
        ],
        "page_size": 20,
        "current_page": 1
    },
    "total_count": 0
}
```

3. Remove in admin configurations Default Page Size value:
    - Go to the `Stores->Configuration->Services->Magento Web API->Web Api Input Limits and clear the Default Page Size value`
    - Use the same API endpoint without "page_size" param:
So request with the API endpoint `rest/all/V1/orders/?searchCriteria[filterGroups][0][filters][0][field]=status&searchCriteria[filterGroups][0][filters][0][value]=processing&searchCriteria[filterGroups][0][filters][0][conditionType]=eq&searchCriteria[currentPage]=1`
 must return response like this:

```
{
    "items": [],
    "search_criteria": {
        "filter_groups": [
            {
                "filters": [
                    {
                        "field": "status",
                        "value": "processing",
                        "condition_type": "eq"
                    }
                ]
            }
        ],
        "page_size": 999999,
        "current_page": 1
    },
    "total_count": 0
}
```

(how we can see, "page_size" = 999999, because in `app/etc/di.xml` for the class Magento\Framework\Webapi\ServiceInputProcessor we have argument defaultPageSize, which setted to 999999)

4. Use the same API endpoint but with "page_size" param (to check that API retrieve custom params correctly):
So request with the API endpoint `rest/all/V1/orders/?searchCriteria[filterGroups][0][filters][0][field]=status&searchCriteria[filterGroups][0][filters][0][value]=processing&searchCriteria[filterGroups][0][filters][0][conditionType]=eq&searchCriteria[currentPage]=1&searchCriteria[pageSize]=20` must return response like this:

```
{
    "items": [],
    "search_criteria": {
        "filter_groups": [
            {
                "filters": [
                    {
                        "field": "status",
                        "value": "processing",
                        "condition_type": "eq"
                    }
                ]
            }
        ],
        "page_size": 20,
        "current_page": 1
    },
    "total_count": 0
}
```

### Questions or comments

### Contribution checklist (*)
